crates/bins/src/bin/datadog_static_analyzer_server/ide/configuration_file/endpoints.rs

```diff
@@ -0,0 +1,63 @@
+use super::error::ConfigFileError;
+use super::models::{AddRuleSetsRequest, IgnoreRuleRequest};
+use super::static_analysis_config_file::StaticAnalysisConfigFile;
+use kernel::utils::encode_base64_string;
+use rocket::http::Status;
+use rocket::serde::json::Json;
+use tracing::instrument;
+
+#[instrument()]
+#[rocket::post(
+    "/v1/config/ignore-rule",
+    format = "application/json",
+    data = "<request>"
+)]
+pub fn ignore_rule(
+    request: Json<IgnoreRuleRequest>,
+) -> Result<Json<String>, rocket::response::status::Custom<ConfigFileError>> {
+    let IgnoreRuleRequest {
+        rule,
+        configuration_base64,
+        encoded,
+        ..
+    } = request.into_inner();
+    tracing::debug!(rule, content = &configuration_base64);
+    let result = StaticAnalysisConfigFile::with_ignored_rule(rule.into(), configuration_base64);
+    to_json_result(result, encoded)
+}
+
+#[instrument()]
+#[rocket::post("/v1/config/rulesets", format = "application/json", data = "<request>")]
+pub fn post_rulesets(
+    request: Json<AddRuleSetsRequest>,
+) -> Result<Json<String>, rocket::response::status::Custom<ConfigFileError>> {
+    let AddRuleSetsRequest {
+        rulesets,
+        configuration_base64,
+        encoded,
+        ..
+    } = request.into_inner();
+    tracing::debug!(
+        rulesets=?&rulesets,
+        content=&configuration_base64
+    );
+    let result = StaticAnalysisConfigFile::with_added_rulesets(&rulesets, configuration_base64);
+    to_json_result(result, encoded)
+}
+
+#[instrument()]
+#[rocket::get("/v1/config/rulesets/<content>", format = "application/json")]
+pub fn get_rulesets(content: &str) -> Json<Vec<String>> {
+    tracing::debug!(%content);
+    Json(StaticAnalysisConfigFile::to_rulesets(content.to_string()))
+}
+
+fn to_json_result(
```

Comment:
Not for this PR, but in a future, we should improve this so it doesn't re-serialize the whole file. Since this is a file that the users might modify manually, by re-writing it we are going to be overwriting comments, specific formatting, the order of the entries, etc. We should be able to preserve that.